docs/concepts/validators.md

```diff
@@ -1,635 +1,575 @@
+           even if they don't make use of it.
 
-`mode='after'` validators are instance methods and always receive an instance of the model as the first argument. Be sure to return the instance at the end of your validator.
-You should not use `(cls, ModelType)` as the signature, instead just use `(self)` and let type checkers infer the type of `self` for you.
-Since these are fully type safe they are often easier to implement than `mode='before'` validators.
-If any field fails to validate, `mode='after'` validators for that field will not be called.
+!!! note "Validation of default values"
+    As mentioned in the [fields documentation](./fields.md#validate-default-values), default values of fields
+    are *not* validated by default, and thus custom validators will not be applied as well.
 
-## Handling errors in validators
+### Using the annotated pattern or the decorators
 
-As mentioned in the previous sections you can raise either a `ValueError` or `AssertionError` (including ones generated by `assert ...` statements) within a validator to indicate validation failed.
-You can also raise a `PydanticCustomError` which is a bit more verbose but gives you extra flexibility.
-Any other errors (including `TypeError`) are bubbled up and not wrapped in a `ValidationError`.
+While both approaches can achieve the same thing, each implementation provide different benefits.
+
+#### Using the annotated pattern
+
+One of the key benefits of using the [annotated pattern](./fields.md#the-annotated-pattern) is to make
+validators reusable:
 
 ```python
-from pydantic_core import PydanticCustomError
+from typing import List
 
-from pydantic import BaseModel, ValidationError, field_validator
+from typing_extensions import Annotated
 
+from pydantic import AfterValidator, BaseModel
 
-class Model(BaseModel):
-    x: int
 
-    @field_validator('x')
-    @classmethod
-    def validate_x(cls, v: int) -> int:
-        if v % 42 == 0:
-            raise PydanticCustomError(
-                'the_answer_error',
-                '{number} is the answer!',
-                {'number': v},
-            )
-        return v
+def is_even(value: int) -> int:
+    if value % 2 == 1:
+        raise ValueError(f'{value} is not an even number')
+    return value
+
+
+EvenNumber = Annotated[str, AfterValidator(is_even)]
 
 
-try:
-    Model(x=42 * 2)
-except ValidationError as e:
-    print(e)
-    """
-    1 validation error for Model
-    x
-      84 is the answer! [type=the_answer_error, input_value=84, input_type=int]
-    """
+class Model1(BaseModel):
+    my_number: EvenNumber
+
+
+class Model2(BaseModel):
+    other_number: Annotated[EvenNumber, AfterValidator(lambda v: v + 2)]
+
+
+class Model3(BaseModel):
+    list_of_even_numbers: List[EvenNumber]  # (1)!
 ```
 
-## Special Types
+1. As mentioned in the [annotated pattern](./fields.md#the-annotated-pattern) documentation,
```

Comment:
I got a bit lost here at first, maybe specify, in this case, the list items, but not the whole list or something like that.